js/src/evaluation/evaluator.ts

```diff
@@ -110,6 +110,21 @@ export class DynamicRunEvaluator<Func extends (...args: any[]) => any>
       langSmithRunAndExample: { run: Run; example: Example };
     }) => {
       const { run, example } = input.langSmithRunAndExample;
+
+      // Check if the evaluator expects the new argument format
+      const params = getEvaluatorParameters(evaluator);
```

Comment:
I think the better/safer way to do this is to shim it with no source code inspection:

```
return evaluator({
         ...run,
          run,
          example,
          inputs: example?.inputs,
          outputs: run?.outputs,
          referenceOutputs: example?.outputs,,
        }, example);
```

Convention in JS is to ignore extra kwargs and args so it should work without breaking anyone.

Downside is it makes logging a little bit funkier but we can deprecate the old style(?)

Reply:
CC @nfcampos I think I heard you should try to avoid `Function.toString()` but I'm not actually sure why (other than the fact that the regexes for some of the below checks should be tweaked)

Perf probably isn't super important given this will only run once

Reply:
Googling around and tinkering in my terminal - implementation seems unclear, I think it may have some funny edge cases (at the very least arrow functions would definitely need a separate regex, along with `async`, maybe others I'm not thinking about)

Would prefer to do something like the shim above in order to avoid any potential gotchas